intro/integration/js.md

```diff
@@ -23,44 +29,141 @@ import skygear from 'skygear/react-native'
 
-Webpack does not work well with Skygear out of the box, if you are building with Webpack, please modify your Webpack config, and add `react-native` to `externals`.
+To use the service in the component, you have to pass it as a provider of the module.
 
 ```
-  externals: [
-    'react-native'
-  ]
+import { SkygearService } from './skygear.service';
+
+@NgModule({
+  declarations: [
+    AppComponent
+  ],
+  imports: [
+    BrowserModule,
+    FormsModule,
+    HttpModule
+  ],
+  providers: [SkygearService],  // This line is want you need
+  bootstrap: [AppComponent]
+})
+export class AppModule {
+}
+
 ```
 
-## Ionic
+Next, accepts the service in the component by modifying the constructor. This creates a private property `skygearService` in the component and assign the service to it.
+
+```
+constructor(private skygearService: SkygearService) {}
+```
 
-If you are using `ionic start` to create an Ionic project, you have to customize its internal Webpack config to get it working with Skygear.
+Next, it is recommended to do necessary configuration in the `ngOnInit` life cycle funcion.
 
-1. update `package.json`, specify your own Webpack config.
 ```
-  "config": {
-    "ionic_webpack": "./webpack.config.json"
+  title = 'Loading...';
+  skygear = null;
+  ngOnInit(): void {
+    this.skygearService.getSkygear()
+    .then(skygear=> {
+      this.skygear = skygear;
+      this.title = "Configurated";
+    })
+    .catch(error=> {
+      this.title = "Cannot configure skygear";
+    });
   }
 ```
 
-2. Create the webpack config file
+Finally, the state can be displayed via the template.
+
 ```
-  {
-    "externals": ["react-native"],
-    "output": {
-      "libraryTarget": "umd",
-      "path": "{{BUILD}}",
-      "publicPath": "build/",
-      "filename": "main.js"
-    }
+<h1>{{title}}</h1>
+```
+
+## Creating your first record with user interaction
+
+First, you need a button, add the following code to your template file, for the default `AppModule`, it will be `app.component.html`.
+
+```
+<button (click)="addNewRecord()">Add new record</button>
+```
+
+Then, you have to add a handler that will be triggered when the button is pressed, it looks like this:
+
+```
+export class AppComponent {
+  addNewRecord() {
+    // Called when button is clicked
   }
+}
 ```
 
-## Angular
+You can then place code to create a record in the handler, and reflect the state to user interface.
 
-If you are using `ng new` to bootstrap your app, you will find it uses Webpack internally to build the project. Unfortunately, there is no way to customize the build process and therefore not possible to make it work with Skygear. We recommend using [Angular-webpack](https://github.com/preboot/angular-webpack) for bootstrapping, and then follow the above guide for integrating with Webpack.
+```
+this.skygearService.getSkygear()
+// Skygear requires a user before creating a record
+.then(()=> this.skygear.signupAnonymously())
```

Comment:
Where does this `this.skygear` came from?

In addition, may we move this signupAnonymously to the `ngOnInit `?